instant-xml-macros/src/lib.rs

```diff
@@ -6,15 +6,43 @@ use syn::parse_macro_input;
 
 #[proc_macro_derive(ToXml)]
 pub fn to_xml(input: TokenStream) -> TokenStream {
-    let ast = parse_macro_input!(input as syn::ItemStruct);
+    let ast = parse_macro_input!(input as syn::DeriveInput);
     let ident = &ast.ident;
-    let name = ident.to_string();
+    let root_name = ident.to_string();
+    let mut output: proc_macro2::TokenStream =
+        TokenStream::from(quote!(::std::format!("<{}>", #root_name)
+            .as_str()
+            .to_owned()))
+        .into();
+
+    match &ast.data {
+        syn::Data::Enum(_) => todo!(),
+        syn::Data::Struct(ref data) => {
+            match data.fields {
+                syn::Fields::Named(ref fields) => {
+                    fields
+                    .named
+                    .iter()
+                    .for_each(|field| {
+                        let field_name = field.ident.as_ref().unwrap().to_string();
+                        let field_value = field.ident.as_ref().unwrap();
+                        output.extend(quote!(+ ::std::format!("<{}>{}</{}>", #field_name, self.#field_value, #field_name).as_str()));
```

Comment:
Doing one `format!()` call per field and concatenating them is very inefficient. I think we should do something better? `ToXml::write_xml()` takes a `fmt::Write`, so we should just write into that directly.

Reply:
Good call, done